utils/utils.go

```diff
@@ -238,20 +237,16 @@ func MarshalAndWriteTable(writer io.Writer, in interface{}) error {
 }
 
 func PrintDetail(writer io.Writer, outputFormat string, list interface{}) error {
-	obj := list
```

Comment:
As this is used for more than just `dapr list`, any concern about the impact of this change on other uses (e.g. component output)?

Reply:
I was working on a [PR of my own](https://github.com/philliphoff/cli/tree/philliphoff-json-lists), mostly as practice in Go, but you beat me to it. :-)

I was concerned about backward compatibility, so my change selectively turned off list unwrapping only for JSON output of `dapr list`.

Reply:
It's used by `dapr components -k` and `dapr configurations -k` to list components & configuration using JSON & YAML outputs, and these commands also had the same inconsistency issue. This change fixes them as well.

Reply:
You'd know better than me. :-)

Reply:
> I was working on a [PR of my own](https://github.com/philliphoff/cli/tree/philliphoff-json-lists), mostly as practice in Go, but you beat me to it. :-)

<details>
  <summary>Haha, apologies!</summary>
  I was motivated to create this PR, hint: look at the PR number and my username :P
</details>

I think it should be okay to start being consistent in all the three outputs (`list`, `components`, `configurations`) and change the behavior. As you mentioned, we can explicitly call it out in the release notes. 

/fyi @mukundansundar

Reply:
The only question that I have here is for yaml should we print in the old format itself without the `-` that I see being added in the tests

Reply:
Isn't this the same analogy as JSON? i.e. printing an object vs printing a list with the single object?